### ISSUE TITLE
500 error getting person page

### ISSUE THREAD

**Issue by bakert** [PennyDreadfulMTG/Penny-Dreadful-Tools#7172]
```
Apr 04 17:05:37 katelyngigante uwsgi[620]: [2020-04-04 17:05:37,143] ERROR in app: Exception on /people/3948/ [GET]
Apr 04 17:05:37 katelyngigante uwsgi[620]: Traceback (most recent call last):
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 2446, in wsgi_app
Apr 04 17:05:37 katelyngigante uwsgi[620]:     response = self.full_dispatch_request()
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 1951, in full_dispatch_request
Apr 04 17:05:37 katelyngigante uwsgi[620]:     rv = self.handle_user_exception(e)
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 1820, in handle_user_exception
Apr 04 17:05:37 katelyngigante uwsgi[620]:     reraise(exc_type, exc_value, tb)
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
Apr 04 17:05:37 katelyngigante uwsgi[620]:     raise value
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 1949, in full_dispatch_request
Apr 04 17:05:37 katelyngigante uwsgi[620]:     rv = self.dispatch_request()
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 1935, in dispatch_request
Apr 04 17:05:37 katelyngigante uwsgi[620]:     return self.view_functions[rule.endpoint](**req.view_args)
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "./decksite/cache.py", line 80, in decorated_function
Apr 04 17:05:37 katelyngigante uwsgi[620]:     response = make_response(f(*args, **kwargs))
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "./decksite/controllers/metagame.py", line 71, in person
Apr 04 17:05:37 katelyngigante uwsgi[620]:     view = Person(p, person_cards, person_archetypes, all_archetypes, person_matchups, your_cards, get_season_id())
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "./decksite/views/person.py", line 32, in __init__
Apr 04 17:05:37 katelyngigante uwsgi[620]:     self.displayed_achievements = [{'title': a.title, 'detail': titlecase.titlecase(a.display(self.person))} for a in Achievement.all_achie>
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "./decksite/views/person.py", line 32, in <listcomp>
Apr 04 17:05:37 katelyngigante uwsgi[620]:     self.displayed_achievements = [{'title': a.title, 'detail': titlecase.titlecase(a.display(self.person))} for a in Achievement.all_achie>
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "./decksite/data/achievements.py", line 259, in display
Apr 04 17:05:37 katelyngigante uwsgi[620]:     return self.localised_display(n)
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "./decksite/data/achievements.py", line 350, in localised_display
Apr 04 17:05:37 katelyngigante uwsgi[620]:     return ngettext('1 victory', '%(num)d victories', n)
Apr 04 17:05:37 katelyngigante uwsgi[620]:   File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask_babel/__init__.py", line 575, in ngettext
Apr 04 17:05:37 katelyngigante uwsgi[620]:     return s if not variables else s % variables
Apr 04 17:05:37 katelyngigante uwsgi[620]: ValueError: unsupported format character '?' (0x43f) at index 7
Apr 04 17:05:37 katelyngigante uwsgi[620]: [2020-04-04 17:05:37,192] ERROR in logger: At request path: /people/3948/

```

**Comment by bakert** [comment]
```

Vorpal Buildbot <notifications@github.com> | Vorpal Buildbot <notifications@github.com> | 5:53 AM (6 hours ago) |   |  
-- | -- | -- | -- | --
Vorpal Buildbot <notifications@github.com>
to PennyDreadfulMTG/perf-reports, Subscribed | to PennyDreadfulMTG/perf-reports, Subscribed
to PennyDreadfulMTG/perf-reports, Subscribed


```

**Comment by bakert** [comment]
```
500 error at /people/robbie2540/

500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.

Reported on decksite by logged_out

InternalServerError 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
Stack Trace:


Python traceback

  File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)


Exception_hash: 15c4ac9d81725488a0e577b38f62ab01229a08cf

Request Data

Request Method: GET
Path: /people/robbie2540/?locale=ru
Cookies: ImmutableMultiDict([('__cfduid', 'de05d7c66b693e163569f8745fe0c82081586087390'), ('session', 'eyJsb2NhbGUiOiJwdF9CUiJ9.XonGEw.rqqagftB_gW7RwnQUOtAlAQ6SGA')])
Endpoint: person
View Args: {'person_id': 'robbie2540'}
Person: logged_out
Referrer: None
Request Data: {}
Host: pennydreadfulmagic.com
Accept-Encoding: gzip
Cf-Ipcountry: DE
X-Forwarded-For: 94.130.9.106, 172.68.143.38
Cf-Ray: 57f30daa5fbf9cb1-SJC
X-Forwarded-Proto: https
Cf-Visitor: {"scheme":"https"}
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
User-Agent: Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.com/)
Cookie: __cfduid=de05d7c66b693e163569f8745fe0c82081586087390; session=eyJsb2NhbGUiOiJwdF9CUiJ9.XonGEw.rqqagftB_gW7RwnQUOtAlAQ6SGA
Cf-Connecting-Ip: 94.130.9.106
Cdn-Loop: cloudflare
X-Forwarded-Host: pennydreadfulmagic.com
X-Forwarded-Server: pennydreadfulmagic.com
Connection: Keep-Alive
```

**Comment by bakert** [comment]
```
500 error at /people/kronosoma87/

500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.

Reported on decksite by logged_out

InternalServerError 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
Stack Trace:


Python traceback

  File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)


Exception_hash: 15c4ac9d81725488a0e577b38f62ab01229a08cf

Request Data

Request Method: GET
Path: /people/kronosoma87/?locale=ru
Cookies: ImmutableMultiDict([('__cfduid', 'd4ac52bc5d37a7a0be751d14481fe35721586084097'), ('session', 'eyJsb2NhbGUiOiJwdF9CUiJ9.Xom5Mw.DCmJfcfYypBeLDTcXJCCm0IB4Bk')])
Endpoint: person
View Args: {'person_id': 'kronosoma87'}
Person: logged_out
Referrer: None
Request Data: {}
Host: pennydreadfulmagic.com
Accept-Encoding: gzip
Cf-Ipcountry: DE
X-Forwarded-For: 94.130.9.106, 172.69.22.106
Cf-Ray: 57f2bd311e9dfa24-SJC
X-Forwarded-Proto: https
Cf-Visitor: {"scheme":"https"}
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
User-Agent: Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.com/)
Cookie: __cfduid=d4ac52bc5d37a7a0be751d14481fe35721586084097; session=eyJsb2NhbGUiOiJwdF9CUiJ9.Xom5Mw.DCmJfcfYypBeLDTcXJCCm0IB4Bk
Cf-Connecting-Ip: 94.130.9.106
Cdn-Loop: cloudflare
X-Forwarded-Host: pennydreadfulmagic.com
X-Forwarded-Server: pennydreadfulmagic.com
Connection: Keep-Alive
```

**Comment by bakert** [comment]
```
500 error at /people/j_meka/

500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.

Reported on decksite by logged_out

InternalServerError 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
Stack Trace:


Python traceback

  File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/discord/.pyenv/versions/3.8.2/lib/python3.8/site-packages/flask/app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)


Exception_hash: 15c4ac9d81725488a0e577b38f62ab01229a08cf

Request Data

Request Method: GET
Path: /people/j_meka/?locale=ru
Cookies: ImmutableMultiDict([('__cfduid', 'd76fdf3b26d61e59ff65b8f4d451673bf1586082880'), ('session', 'eyJsb2NhbGUiOiJwdF9CUiJ9.Xom0YA.0us9yHILFWKlR1DK4lb5Rf46ZEY')])
Endpoint: person
View Args: {'person_id': 'j_meka'}
Person: logged_out
Referrer: None
Request Data: {}
Host: pennydreadfulmagic.com
Accept-Encoding: gzip
Cf-Ipcountry: DE
X-Forwarded-For: 94.130.9.106, 172.68.142.169
Cf-Ray: 57f29f08bad9c795-SJC
X-Forwarded-Proto: https
Cf-Visitor: {"scheme":"https"}
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
User-Agent: Mozilla/5.0 (compatible; BLEXBot/1.0; +http://webmeup-crawler.com/)
Cookie: __cfduid=d76fdf3b26d61e59ff65b8f4d451673bf1586082880; session=eyJsb2NhbGUiOiJwdF9CUiJ9.Xom0YA.0us9yHILFWKlR1DK4lb5Rf46ZEY
Cf-Connecting-Ip: 94.130.9.106
Cdn-Loop: cloudflare
X-Forwarded-Host: pennydreadfulmagic.com
X-Forwarded-Server: pennydreadfulmagic.com
Connection: Keep-Alive
```

**Comment by bakert** [comment]
(there are more)

**Comment by bakert** [comment]
``
Mar 21 09:16:27 katelyngigante uwsgi[982]: [2020-03-21 09:16:27,687] ERROR in app: Exception on /people/byvci/ [GET]
Mar 21 09:16:27 katelyngigante uwsgi[982]: Traceback (most recent call last):
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
Mar 21 09:16:27 katelyngigante uwsgi[982]:     response = self.full_dispatch_request()
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
Mar 21 09:16:27 katelyngigante uwsgi[982]:     rv = self.handle_user_exception(e)
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
Mar 21 09:16:27 katelyngigante uwsgi[982]:     reraise(exc_type, exc_value, tb)
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
Mar 21 09:16:27 katelyngigante uwsgi[982]:     raise value
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
Mar 21 09:16:27 katelyngigante uwsgi[982]:     rv = self.dispatch_request()
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
Mar 21 09:16:27 katelyngigante uwsgi[982]:     return self.view_functions[rule.endpoint](**req.view_args)
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "./decksite/cache.py", line 80, in decorated_function
Mar 21 09:16:27 katelyngigante uwsgi[982]:     response = make_response(f(*args, **kwargs))
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "./decksite/controllers/metagame.py", line 71, in person
Mar 21 09:16:27 katelyngigante uwsgi[982]:     view = Person(p, person_cards, person_archetypes, all_archetypes, person_matchups, your_cards, get_season_id())
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "./decksite/views/person.py", line 32, in __init__
Mar 21 09:16:27 katelyngigante uwsgi[982]:     self.displayed_achievements = [{'title': a.title, 'detail': titlecase.titlecase(a.display(self.person))} for a in Achievement.all_achievements if a.display(self.person)]
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "./decksite/views/person.py", line 32, in <listcomp>
Mar 21 09:16:27 katelyngigante uwsgi[982]:     self.displayed_achievements = [{'title': a.title, 'detail': titlecase.titlecase(a.display(self.person))} for a in Achievement.all_achievements if a.display(self.person)]
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "./decksite/data/achievements.py", line 259, in display
Mar 21 09:16:27 katelyngigante uwsgi[982]:     return self.localised_display(n)
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "./decksite/data/achievements.py", line 350, in localised_display
Mar 21 09:16:27 katelyngigante uwsgi[982]:     return ngettext('1 victory', '%(num)d victories', n)
Mar 21 09:16:27 katelyngigante uwsgi[982]:   File "/home/discord/.local/lib/python3.6/site-packages/flask_babel/__init__.py", line 575, in ngettext
Mar 21 09:16:27 katelyngigante uwsgi[982]:     return s if not variables else s % variables
Mar 21 09:16:27 katelyngigante uwsgi[982]: ValueError: unsupported format character '?' (0x43f) at index 7
Mar 21 09:16:27 katelyngigante uwsgi[982]: [2020-03-21 09:16:27,689] ERROR in logger: At request path: /people/byvci/
Mar 21 09:16:27 katelyngigante uwsgi[982]: werkzeug.exceptions.InternalServerError: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
```

**Comment by bakert** [comment]
This particular one seems to be choking on the cyrillic character 'п'

**Comment by bakert** [comment]
https://pennydreadfulmagic.com/people/duckoforegon/?locale=ru reproduces the problem

**Comment by bakert** [comment]
Even after copying back the db to local I can't repro on local.

**Comment by bakert** [comment]
The problem here is that the translation has:

```
#: decksite/data/achievements.py:350
#, python-format
msgid "1 victory"
msgid_plural "%(num)d victories"
msgstr[0] "1 победа"
msgstr[1] "%(num) побед"
msgstr[2] "%(num) побед"
```

It should be:

```
#: decksite/data/achievements.py:350
#, python-format
msgid "1 victory"
msgid_plural "%(num)d victories"
msgstr[0] "1 победа"
msgstr[1] "%(num)d побед"
msgstr[2] "%(num)d побед"
```

**Comment by bakert** [comment]
Fixes needed:

 - [x] Change the actual translation
 - [x] validate_translations should not allow formulations like that through.
 - [x] add a test of validate_translations.has_missing_var to check it finds these in future.

**Comment by bakert** [comment]
I changed the translation in poeditor and pulled that onto prod.

Fixes to prevent this in future in 391f30f7.
